sqlserver/datadog_checks/sqlserver/statements.py

```diff
@@ -328,8 +332,12 @@ def _load_plan(self, query_hash, query_plan_hash, cursor):
         return result[0][0]
 
     @tracked_method(agent_check_getter=agent_check_getter)
-    def _collect_plans(self, rows, cursor):
+    def _collect_plans(self, rows, cursor, deadline):
         for row in rows:
+            if self.enforce_collection_interval_deadline and time.time() > deadline:
+                self.log.debug("ending plan collection early because check deadline has been exceeded")
+                self.check.count("dd.sqlserver.statements.deadline_exceeded", 1, **self.check.debug_stats_kwargs())
```

Comment:
Not too familiar with this. Is this an internal metric?

Reply:
Yes we have a variety of internal troubleshooting metrics for this check. All metrics prefixed with `dd.sqlserver` are internal only to the check and not part of the product.